client/src/pages/queries/components/chart-tab.tsx

```diff
@@ -0,0 +1,249 @@
+import React, { useMemo, useState } from 'react'
+import { Empty, Form, Select, Input } from 'antd'
+import { Line, Area, Column, Scatter, Bar, Pie } from '@ant-design/charts'
+import {
+  LineChartOutlined,
+  AreaChartOutlined,
+  PieChartOutlined,
+  BarChartOutlined,
+  DotChartOutlined,
+} from '@ant-design/icons'
+import { QueryResult } from 'types/query'
+
+type ChartTabProps = {
+  data: QueryResult | null
+}
+
+export default function ChartTab({ data }: ChartTabProps) {
+  const [label, setLabel] = useState<string | undefined>()
+  const [values, setValues] = useState<string | string[] | undefined>()
+  const [title, setTitle] = useState<string | undefined>('Untitled Chart')
+  const [chartType, setChartType] = useState<string>('line')
+
+  const chartData = useMemo(() => {
+    if (label && values && data) {
+      if (typeof values === 'string') {
+        return data.rows.map((row) => {
+          return { label: row[label], value: row[values], type: values }
+        })
+      } else {
+        return values
+          .map((value) =>
+            data.rows.map((row) => {
+              return { label: row[label], value: row[value], type: value }
+            }),
+          )
+          .flat()
+      }
+    } else {
+      return []
+    }
+  }, [label, values, data])
+
+  const config = useMemo(() => {
+    return {
+      data: chartData,
+      xField: 'label',
+      yField: 'value',
+      seriesField: 'type',
+      autoFit: true,
+    }
+  }, [chartData])
+
+  const chartList = useMemo(
```

Comment:
`chartList` can be moved out of the component like

```javascript
const chartTypes = {
  line: {
    icon: <LineCartOutlined />,
    label: 'Line Chart',
    chart: <LineChart />,
  }
}

const chartsOrder = [
  'line', 'area', 'stacked-area',
]
```

`chartsOrder` can be used to `map` like

```javascript
chartsOrder.map(chartId => <Select.Option key={chartId}>{chartTypes[chartId].label}</Select.Option>)
```

and the `chart` component can be rendered as

```javascript
cloneElement(chartType[chartId].chart, { ...config, data: chartData })
```